plugin/src/main/java/dev/regadas/trino/pubsub/listener/Encoder.java

```diff
@@ -29,21 +31,12 @@ public static Encoding fromOrDefault(String encoding, Encoding defaultEncoding)
 
     @FunctionalInterface
     interface MessageEncoder extends Encoder<Message> {
-        static final JsonFormat.Printer JSON_PRINTER = JsonFormat.printer();
+        JsonFormat.Printer JSON_PRINTER = JsonFormat.printer();
 
         static MessageEncoder create(Encoding encoding) {
-            return new MessageEncoder() {
-                @Override
-                public byte[] encode(Message value) throws Exception {
-                    switch (encoding) {
-                        case JSON:
-                            return JSON_PRINTER.print(value).getBytes("UTF-8");
-                        case PROTO:
-                            return value.toByteArray();
-                        default:
-                            throw new IllegalArgumentException("Unknown encoding: " + encoding);
-                    }
-                }
+            return switch (encoding) {
+                case JSON -> value -> JSON_PRINTER.print(value).getBytes(UTF_8);
+                case PROTO -> MessageLite::toByteArray;
```

Comment:
```suggestion
                case PROTO -> Message::toByteArray;
```

Since we have `Message` already in scope there's no benefit here.